extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.tracking.js

```diff
@@ -1,35 +1,152 @@
 define('wikia.articleVideo.featuredVideo.tracking', [], function () {
+	var state = getDefaultState();
+	var gaCategory = 'featured-video';
+	var playerInstance;
+	var wasAlreadyUnmuted = false;
+	var depth = 0;
+
 	function getDefaultState() {
 		return {
-			wasAlreadyUnmuted: false
+			wasStartTracked: false,
+			progress: {
+				durationTracked: 0,
+				percentTracked: 0
+			}
 		}
 	}
 
 	function track(gaData) {
-		console.log(gaData);
+		if (!gaData.label) {
+			throw new Error('No tracking label provided');
+		}
+
+		var finalGAData = {
+			action: gaData.action || 'click',
+			category: gaCategory,
+			label: gaData.label,
+			//value tracks sound state: 1 for muted, 0 for unmuted
+			value: Number(playerInstance.getMute())
+		};
+
+		// Will be replaced by connecting Internal + GA trackers
+		console.info(gaData);
 	}
 
-	var state = getDefaultState();
+	return function (providedPlayerInstance, willAutoplay, gaCategory) {
+		playerInstance = providedPlayerInstance;
+		gaCategory = gaCategory;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove or correct this useless self-assignment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1656)